### PR TITLE
Array binary search

### PR DESCRIPTION
**Pull request by paronicholas**
- [x] Top-level README “Table of Contents” is updated
- [x] Feature tasks for this challenge are completed
- [x] Unit tests written and passing
    - [x] “Happy Path” - Expected outcome
    - [x] Expected failure
    - [x] Edge Case (if applicable/obvious)
- [x] README for this challenge is complete
    - [x] Summary, Description, Approach & Efficiency, Solution
    - [x] Link to code
    - [x] Picture of whiteboard

### CODE REVIEW

File: code401challenges/src/main/java/code401challenges/BinarySearch.java [hunk 1]
```diff
@@ -2,6 +2,16 @@
 
 public class BinarySearch {
   public static int binarySearch(int[] arr, int n) {
-    return 0;
+    int begin = 0;
+    int end = arr.length - 1;
+
+    while (begin <= end) {
+      int mid = (int)Math.ceil(((double)begin + (double)end) / 2);
```
Comment (evanslaton) [inline code401challenges/src/main/java/code401challenges/BinarySearch.java]: Each time this loops `mid` is declared and assigned a value. If you declare `mid` outside the loop and just reassign `mid` inside the loop, your loop is doing a little less work. The efficiency of this is negligible but it's worth thinking about!
